internal/integration/csot_prose_test.go

```diff
@@ -83,6 +84,126 @@ func TestCSOTProse(t *testing.T) {
+		cliOptions := options.Client().SetTimeout(10).ApplyURI(mtest.ClusterURI())
+		integtest.AddTestServerAPIVersion(cliOptions)
+
+		client, err := mongo.Connect(cliOptions)
+		assert.NoError(t, err, "failed to connect to server")
+
+		// Create a GridFS bucket that wraps the db database.
+		bucket := client.Database("db").GridFSBucket()
+
+		// Note that UploadFromStream accounts for the following steps:
+		// - Call bucket.open_upload_stream()
+		// - Using uploadStream, upload a single 0x12 byte
+		// - Call uploadStream.close() to flush the stream and insert chunks
+		_, err = bucket.UploadFromStream(context.Background(), "filename", bytes.NewReader([]byte{0x12}))
+		assert.ErrorIs(t, err, context.DeadlineExceeded)
+	})
+
+	const test61 = "6.1 gridfs - upload and download with non-expiring client-level timeout"
+	mt.RunOpts(test61, mtest.NewOptions().MinServerVersion("4.4"), func(mt *mtest.T) {
+		// Drop and re-create the db.fs.files and db.fs.chunks collections.
+		err := mt.Client.Database("db").Collection("fs.files").Drop(context.Background())
+		assert.NoError(t, err, "failed to drop files")
+
+		err = mt.Client.Database("db").Collection("fs.chunks").Drop(context.Background())
+		assert.NoError(t, err, "failed to drop chunks")
+
+		// Create a new MongoClient with timeoutMS=10.
+		cliOptions := options.Client().SetTimeout(500 * time.Millisecond).ApplyURI(mtest.ClusterURI())
+		integtest.AddTestServerAPIVersion(cliOptions)
+
+		client, err := mongo.Connect(cliOptions)
+		assert.NoError(t, err, "failed to connect to server")
+
+		// Create a GridFS bucket that wraps the db database.
+		bucket := client.Database("db").GridFSBucket()
+
+		// Upload file and ensure it uploaded correctly.
+		fileID, err := bucket.UploadFromStream(context.Background(), "filename", bytes.NewReader([]byte{0x12}))
+		assert.NoError(t, err, "failed to upload stream")
+
+		buf := bytes.Buffer{}
+
+		_, err = bucket.DownloadToStream(context.Background(), fileID, &buf)
+		assert.NoError(t, err, "failed to download stream")
+		assert.Equal(t, buf.Len(), 1)
+		assert.Equal(t, buf.Bytes(), []byte{0x12})
+	})
+
+	const test62 = "6.2 gridfs - upload with operation-level timeout"
+	mt.RunOpts(test62, mtest.NewOptions().MinServerVersion("4.4"), func(mt *mtest.T) {
+		// Drop and re-create the db.fs.files and db.fs.chunks collections.
+		err := mt.Client.Database("db").Collection("fs.files").Drop(context.Background())
+		assert.NoError(t, err, "failed to drop files")
+
+		err = mt.Client.Database("db").Collection("fs.chunks").Drop(context.Background())
+		assert.NoError(t, err, "failed to drop chunks")
+
+		// Set a blocking "insert" fail point.
+		mt.SetFailPoint(mtest.FailPoint{
+			ConfigureFailPoint: "failCommand",
+			Mode: mtest.FailPointMode{
+				Times: 1,
+			},
+			Data: mtest.FailPointData{
+				FailCommands:    []string{"insert"},
+				BlockConnection: true,
+				BlockTimeMS:     15,
+			},
+		})
+
+		// Create a new MongoClient with timeoutMS=10.
```

Comment:
It will be better to update the comment to describe the code more accurately.

Reply:
Do you have something in mind? This comment is taken from the prose test: https://github.com/mongodb/specifications/tree/master/source/client-side-operations-timeout/tests#6-gridfs---upload

Reply:
Never mind, it looks good now.

Reply:
I may have missed something, but is the "timeoutMS=10" still accurate after https://github.com/mongodb/mongo-go-driver/pull/1782/commits/bf50559321fed60faeaac4528e8c47ce7199207b? Where is the timeout set?